src/HttpServer.jl

```diff
@@ -89,14 +89,24 @@ handler.events["foo"] = (bar) "Hello \$bar"
 HttpServer.event(server, "foo", "Julia")
 ```
 """
+
 immutable HttpHandler
     handle::Function
     sock::Base.TCPServer
     events::Dict
 
-    HttpHandler(handle::Function, sock::Base.TCPServer) = new(handle, sock, defaultevents)
-    HttpHandler(handle::Function) = new(handle, Base.TCPServer(), defaultevents)
+    function HttpHandler(handle::Function, sock::Base.TCPServer)
+        try
```

Comment:
It's possible that the user passes in a generic function to do the HttpHandler constructor, which at the time doesn't have a two-argument method defined, but the user might define a two-argument method on that function  before the server is actually run. So this code's logic isn't perfectly backwards-compatible.